### PR TITLE
Updated Overview and Comparison Modal

### PR DESCRIPTION
**Pull request by vickiwong85**
I was able to pass product data into overview to show the product info, main photo, and thumbnails. It should update correctly when clicked on a new related item and the main product is set to a new product. 

Also edited some of the CSS to match the requirements in the business docs/example page. Added a description and features section (since I have to match the features in the comparisons modal).

For the comparisons modal, I was able to combine and filter the features of both the current product and related product they clicked on and it show show as columns in the modal table. Still have to work on getting checkmarks to show up.

### CODE REVIEW

File: src/components/Comparisons/Comparisons.jsx [hunk 1]
```diff
@@ -46,13 +46,17 @@ function Comparisons (props) {
...
 
-  // gets related products for the current product
+  // gets related products for the current product and removes any duplicates and if it also contains itself as related
   useEffect(() => {
     if (props.prod) {
       var finalArray = [];
       server.get('/products/related', {product_id: props.prod.id})
       .then((data) => {
         var relatedProducts = data.data;
+        if (relatedProducts.includes(props.prod.id)) {
```
Comment (kjmoyer) [inline src/components/Comparisons/Comparisons.jsx:56]: Nice solution to duplicates!

File: src/components/Comparisons/Comparisons.jsx [hunk 2]
```diff
@@ -65,57 +69,61 @@ function Comparisons (props) {
...
-  //66643 seems to be out of stock... so may have to check if there are any in stock first because thumbnail and other properties are null
+  // works but if you click on too many items it'll max out the requests to the API... don't click on too many related products in a short time fram.
+  useEffect(() => {
+    if (products) {
+      related.forEach((item) => {
+        server.get('/products/styles', {product_id: item})
+        .then((data)=> {
+          if (data.data.results[0].skus.null) {
+            if (data.data.results[0].skus.null.quantity === null) {
+              console.log(`${item} is out of stock`);
```
Comment (kjmoyer) [inline src/components/Comparisons/Comparisons.jsx]: Not sure if this is needed? If we need to alert the user, a console log won't tell them - but probably just as effective if it doesn't show up? Not sure if the BRDs say something specific. Just a thought for reducing console.logs

File: src/components/Comparisons/Comparisons.jsx [hunk 3]
```diff
@@ -65,57 +69,61 @@ function Comparisons (props) {
...
+              var index = related.indexOf(item);
+              var copy = related.slice();
+              copy.splice(index, 1);
+              setRelated(copy);
+            }
+            console.log('all related items in stock')
```
Comment (kjmoyer) [inline src/components/Comparisons/Comparisons.jsx]: Same thought here about consolelogs

File: src/components/Comparisons/Comparisons.jsx [hunk 4]
```diff
@@ -124,7 +132,7 @@ function Comparisons (props) {
   function scrollL () {
     console.log('clicked on left button')
```
Comment (kjmoyer) [inline src/components/Comparisons/Comparisons.jsx]: Kill this one :)

File: src/components/Comparisons/Comparisons.jsx [hunk 5]
```diff
@@ -133,7 +141,7 @@ function Comparisons (props) {
     console.log('clicked on right button')
```
Comment (kjmoyer) [inline src/components/Comparisons/Comparisons.jsx]: And this one!

File: src/components/Comparisons/ComparisonsModal.jsx [hunk 6]
```diff
@@ -41,11 +41,30 @@ const TL = styled.td`
...
   }
+  const [characteristics, setCharacteristics] = useState([]);
+
+  //props.prod = current product being viewed
+  //props.item = related item clicked on
+
+  //sets characteristics to the features of both the current product being viewed and the related item the user clicked on (with no duplicates)
+  useEffect(() => {
+    var combinedChar = props.prod.features.concat(props.item.features);
+    console.log(combinedChar);
```
Comment (kjmoyer) [inline src/components/Comparisons/ComparisonsModal.jsx]: More console.logs!

File: src/components/Comparisons/ComparisonsModal.jsx [hunk 7]
```diff
@@ -41,11 +41,30 @@ const TL = styled.td`
...
+    const key = 'feature';
+    const unique = [...new Map(combinedChar.map(item =>
+      [item[key], item])).values()];
+    console.log(unique);
```
Comment (kjmoyer) [inline src/components/Comparisons/ComparisonsModal.jsx]: And another!

File: src/components/Comparisons/RelatedItem.jsx [hunk 8]
```diff
@@ -90,6 +100,7 @@ function RelatedItem (props) {
     }
   }, [reviews])
 
+
   function showModal () {
     console.log('Clicked Star Button');
```
Comment (kjmoyer) [inline src/components/Comparisons/RelatedItem.jsx]: Kill this one :)

File: src/components/Comparisons/ComparisonsModal.jsx [hunk 9]
```diff
@@ -41,11 +41,30 @@ const TL = styled.td`
...
     return null;
   }
+  const [characteristics, setCharacteristics] = useState([]);
+
+  //props.prod = current product being viewed
+  //props.item = related item clicked on
+
+  //sets characteristics to the features of both the current product being viewed and the related item the user clicked on (with no duplicates)
+  useEffect(() => {
+    var combinedChar = props.prod.features.concat(props.item.features);
```
Comment (kjmoyer) [inline src/components/Comparisons/ComparisonsModal.jsx:58]: Somewhere where you concat these together, it seems to be leaving out spaces which you can see in the modal. Need to find where that's happening and adjust (for now maybe just add it as a refactoring ticket?)

File: src/components/Comparisons/RelatedItem.jsx [hunk 10]
```diff
@@ -44,14 +44,24 @@ function RelatedItem (props) {
...
-          var thumbnail = data.data.results[0].photos[0].thumbnail_url
-          setImage(thumbnail);
+          var results = data.data.results;
+          for (var i = 0; i < results.length; i++) {
+            if (results[i]['default?'] === true) {
+              thumbnail = results[i].photos[0].thumbnail_url;
+              setImage(thumbnail);
+            }
+          }
+          if (thumbnail === '') {
```
Comment (kjmoyer) [inline src/components/Comparisons/RelatedItem.jsx:61]: Curious why this is needed? Do some items not have a listed default I'm guessing? (asking just for curiousity's sake)

**Review by kjmoyer — CHANGES_REQUESTED** [review]
Could you clean up the console.logs? :) I put a few other comments, but I think just the console.logs really are needed before merging.

**Review by agonzalezengr — APPROVED** [review]
Impressive CSS styling! I wonder if there is any way to include all that code (ImageGallery.JSX - lines 4 through 26) into a separate file? If not, then no problem! Great work!

File: src/components/Comparisons/ComparisonsModal.jsx (hunk 9 above)
Comment (vickiwong85) [reply src/components/Comparisons/ComparisonsModal.jsx:58]: Waittt how are you able to see this without running the code? haha 

So I'm not sure why but the API sends back some of the values with no spaces between the words.. But yes will have to edit that in a new ticket...

Screen Shot 2022-09-01 at 4 32 26 PM

File: src/components/Comparisons/RelatedItem.jsx (hunk 10 above)
Comment (vickiwong85) [reply src/components/Comparisons/RelatedItem.jsx:61]: Yes. Product 66648 has no styles set as default?: true. they're all in stock but all say default: false. Not sure why!

File: src/components/Comparisons/Comparisons.jsx (hunk 2 above)
Comment (vickiwong85) [reply src/components/Comparisons/Comparisons.jsx]: Done!

**Comment by vickiwong85** [comment]
Removed all console logs I believe!

**Review by kjmoyer — APPROVED** [review]

**Pull request merged**